pkg/controller/controller_binding.go

```diff
@@ -577,108 +584,204 @@ func (c *controller) reconcileServiceBinding(binding *v1beta1.ServiceBinding) er
-			originatingIdentity, err := buildOriginatingIdentity(binding.Spec.UserInfo)
+			toUpdate, err = c.recordStartOfServiceBindingOperation(toUpdate, v1beta1.ServiceBindingOperationUnbind)
 			if err != nil {
-				s := fmt.Sprintf(`Error building originating identity headers while unbinding: %v`, err)
-				glog.Warning(pcb.Message(s))
-				c.recorder.Event(binding, corev1.EventTypeWarning, errorWithOriginatingIdentity, s)
-				setServiceBindingCondition(
-					toUpdate,
-					v1beta1.ServiceBindingConditionReady,
-					v1beta1.ConditionFalse,
-					errorWithOriginatingIdentity,
-					s,
-				)
-				if _, err := c.updateServiceBindingStatus(toUpdate); err != nil {
-					return err
-				}
+				// There has been an update to the binding. Start reconciliation
+				// over with a fresh view of the binding.
 				return err
 			}
-			unbindRequest.OriginatingIdentity = originatingIdentity
 		}
+	}
 
-		if toUpdate.DeletionTimestamp == nil {
-			if toUpdate.Status.OperationStartTime == nil {
-				now := metav1.Now()
-				toUpdate.Status.OperationStartTime = &now
-			}
-		} else {
-			if toUpdate.Status.CurrentOperation != v1beta1.ServiceBindingOperationUnbind {
-				// Cancel any pending orphan mitigation since the resource is being deleted
-				toUpdate.Status.OrphanMitigationInProgress = false
-
-				toUpdate, err = c.recordStartOfServiceBindingOperation(toUpdate, v1beta1.ServiceBindingOperationUnbind)
-				if err != nil {
-					// There has been an update to the binding. Start reconciliation
-					// over with a fresh view of the binding.
-					return err
-				}
-			}
+	if binding.Status.UnbindStatus == v1beta1.ServiceBindingUnbindStatusRequired {
+		if ok, err := c.serviceBindingRequestUnbinding(binding, toUpdate, pcb); !ok || err != nil {
+			return err
 		}
+	}
+
+	if toUpdate.Status.OrphanMitigationInProgress {
+		s := "Orphan mitigation successful"
+		setServiceBindingCondition(toUpdate,
+			v1beta1.ServiceBindingConditionReady,
+			v1beta1.ConditionFalse,
+			successOrphanMitigationReason,
+			s)
+	} else {
+		s := "The binding was deleted successfully"
+		setServiceBindingCondition(
+			toUpdate,
+			v1beta1.ServiceBindingConditionReady,
+			v1beta1.ConditionFalse,
+			successUnboundReason,
+			s,
+		)
+		// Clear the finalizer
+		finalizers.Delete(v1beta1.FinalizerServiceCatalog)
+		toUpdate.Finalizers = finalizers.List()
+	}
+
+	toUpdate.Status.ExternalProperties = nil
+	clearServiceBindingCurrentOperation(toUpdate)
+	if _, err := c.updateServiceBindingStatus(toUpdate); err != nil {
+		return err
+	}
 
-		_, err = brokerClient.Unbind(unbindRequest)
+	c.recorder.Event(binding, corev1.EventTypeNormal, successUnboundReason, "This binding was deleted successfully")
+	glog.V(5).Info(pcb.Message("Successfully deleted ServiceBinding"))
+
+	return nil
+}
+
+func (c *controller) serviceBindingRequestUnbinding(binding *v1beta1.ServiceBinding, toUpdate *v1beta1.ServiceBinding, pcb *pretty.ContextBuilder) (bool, error) {
```

Comment:
This function needs a comment to, at a minimum, make it clear what the function returns.